pages/blog/fetch-for-servers.mdx

```diff
@@ -0,0 +1,171 @@
+
+One of the most powerful trend in the JS ecosystem was the proliferation of new environments and platforms that can run JS(Lambdas, Cloudflare Workers, Deno, Bun etc.). So we set up to build a single GraphQL server framework that could run on any of these platforms.
+
+What we’ve found in the process was fascinating and we believe would change how any JS HTTP frameworks are being built, without any relation to GraphQL
+
+While Node.js is the most popular environment, multiple platforms can run JavaScript with different APIs. Usually, they have their own way of creating servers, but we saw they share a web standard that most people usually use for communicating with the backends from the client side, not the other way around. That’s WHATWG Fetch API that uses `Request`, `Response`, `ReadableStream`, etc to manage this on the server side, too.
+
+## Why Fetch API?
+
+When you send a request from the client, you basically use `fetch(...requestArgs)` that uses `Request` object under the hood that contains all the details(headers, method, etc) and the data stream you need for that communication, then you take `Response` object that contains the connection stream and it processes it as you want with `.json`, `.arrayBuffer`, `.formData` or you can just access the stream itself by `.body` as `ReadableStream`
+
+On the server side, you can take `Request` object as a server and then process it with the same methods without dealing with the internals of your platform.
+
+Streaming responses like SSE, we just use `ReadableStream` and for Multipart requests (e.g. file uploads), we just take `FormData` which is exactly the same passed from the browser or any other client uses Fetch API.
+
+File Uploads; [https://github.com/dotansimha/graphql-yoga/blob/master/packages/common/src/plugins/requestParser/POSTMultipart.ts#L18](https://github.com/dotansimha/graphql-yoga/blob/master/packages/common/src/plugins/requestParser/POSTMultipart.ts#L18)
+
+SSE; [https://github.com/dotansimha/graphql-yoga/blob/master/packages/common/src/plugins/resultProcessor/push.ts#L42](https://github.com/dotansimha/graphql-yoga/blob/master/packages/common/src/plugins/resultProcessor/push.ts#L42)
+
+## What about Node.js?
+
+Currently, in the older LTS versions of Node.js, we don’t have an implementation of the Fetch API builtin. Furthermore, Node.js doesn’t use Web standard streams and the Fetch API in its `http` and `https` modules. That’s why we created the `@whatwg-node/fetch` package (previously known as `cross-undici-fetch` ) that fills in the gaps of different `fetch` implementations in all the LTS Node.js versions. Under the hood, `@whatwg-node/fetch` utilizes `undici` if available or otherwise falls back to using `node-fetch`, which you are probably already familiar with. In case `@whatwg-node/fetch` is imported in an environment that already has Fetch API built in like Cloudflare Workers, no ponyfills are added to your built application bundle.
```

Comment:
```suggestion
Currently, in the older LTS versions of Node.js, we don’t have an implementation of the Fetch API built in. Furthermore, Node.js doesn’t use Web standard streams and the Fetch API in its `http` and `https` modules. That’s why we created the `@whatwg-node/fetch` package (previously known as `cross-undici-fetch` ) that fills in the gaps of different `fetch` implementations in all the LTS Node.js versions. Under the hood, `@whatwg-node/fetch` utilizes `undici` if available or otherwise falls back to using `node-fetch`, which you are probably already familiar with. In case `@whatwg-node/fetch` is imported in an environment that already has Fetch API built in like Cloudflare Workers, no ponyfills are added to your built application bundle.
```